org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/rule/RuleDetailsPanel.java

```diff
@@ -110,31 +111,38 @@ private void updateScrollCompositeMinSize() {
   }
 
   public void updateRule(GetStandaloneRuleDescriptionResponse getStandaloneRuleDescriptionResponse) {
-    var ruleDefinition = getStandaloneRuleDescriptionResponse.getRuleDefinition();
+    try {
+      var ruleDefinition = getStandaloneRuleDescriptionResponse.getRuleDefinition();
 
-    ruleNameLabel.setText(ruleDefinition.getName());
-    ruleNameLabel.requestLayout();
-    ruleHeaderPanel.updateRule(ruleDefinition.getKey(), ruleDefinition.getType(), ruleDefinition.getDefaultSeverity());
+      ruleNameLabel.setText(ruleDefinition.getName());
+      ruleNameLabel.requestLayout();
+      ruleHeaderPanel.updateRule(ruleDefinition.getKey(), ruleDefinition.getType(), ruleDefinition.getDefaultSeverity());
 
-    updateHtmlDescription(getStandaloneRuleDescriptionResponse.getDescription(), ruleDefinition.getLanguage().getLanguageKey());
+      updateHtmlDescription(getStandaloneRuleDescriptionResponse.getDescription(), ruleDefinition.getLanguage().getLanguageKey());
 
-    requestLayout();
-    updateScrollCompositeMinSize();
+      requestLayout();
+      updateScrollCompositeMinSize();
+    } catch (SWTException ignored) {
+      // There might be a race condition between the background job running late and the view already being closed
+    }
```

Comment:
I'm a bit puzzled by this, it looks strange to me that we have to catch such an exception. Isn't there anything we could do to prevent it from happening?
Same remark for the method below

Reply:
We populate the rule content in a background job triggered e.g. when switching the rule outside of the UI thread. When it takes some time (e.g. on the slow Cirrus CI build and I got it on my private machine) and in the meantime, the view / Settings panel is closed, then the exception is thrown regarding the parent composite already being disposed of.
It is not caught in the background job in its try-catch block as we run the methods via `Display.getDefault().syncExec(() -> RuleDetailsPanel.updateRule(...))`.